middleware/applyOverrides.js

```diff
@@ -0,0 +1,41 @@
+const _ = require('lodash');
+const logger = require('pelias-logger').get('api');
+const codec = require('pelias-model').codec;
+
+function setup() {
+  return applyOverrides;
+}
+
+function applyOverrides(req, res, next) {
+  // do nothing if no result data set
+  if (!res || !res.data) {
+    return next();
+  }
+
+  res.data = res.data.map(overrideOneRecord);
+
+  next();
+}
+
+/*
+ * Rename the fields in one record
+ */
+function overrideOneRecord(place) {
+  if (place.addendum && place.addendum.override) {
+    try {
+      const overrideData = codec.decode(place.addendum.override);
+      place = _.merge(place, overrideData);
```

Comment:
This command is very powerful because it bypasses all the setter logic baked into `pelias/model` which ensures validity and type correctness.

It would be fairly easy to do something silly like put a string where an array was expected for vice versa which would cause a runtime error and return a HTTP 500

Reply:
That's true. I'm not sure how I could easily reuse the pelias-model logic it seems like it doesn't have a concept of loading/validating an object, so I don't know how I could use it at import or serving time to guarantee correctness.

Another approach would be to add a "displayName" property to pelias-model which seems almost more fragile since there would be so many places to decide if I should use name or displayName.

Do you have any better ideas on how this could be implemented?

Reply:
@orangejulius and I just discussed this in our call today.

It's a very powerful tool, so we're happy to merge this as an undocumented feature with the disclaimer that it can break things and errors made in the addendum will require a re-index to fix.

With that in mind I prefer the flexibility of `addendum.override` than `displayName ` as it potentially allows prototyping other features in the future without changing code.